newIDE/app/src/ProjectCreation/LocalExamples.js

```diff
@@ -54,11 +55,14 @@ export const showGameFileCreationError = (
 
 export default class LocalExamples extends Component<Props, State> {
   state = {
-    outputPath: findEmptyPath(
-      path && app
-        ? path.join(app.getPath('documents'), 'GDevelop projects')
-        : ''
-    ),
+    outputPath:
```

Comment:
If you store the outputPath string in the parent component, you must then remove it from the state of this component. Otherwise you're duplicating the state, which comes with several risks if not done properly.

Moving the state from a component to a parent component is quite common in React. It even has a name, it's called "Lifting Up State" because you're moving from a children component to a parent component. Read this nice article about it: https://reactjs.org/docs/lifting-state-up.html

Once you get it, it's a very interesting way of thinking: all in React is about where you put the state, and what prop you give to your children :)